### ISSUE TITLE
Peasants to workers conversion priority

### ISSUE THREAD

**Issue by Rakso69** [Sh0wdown/UnofficialCrusaderPatch#61]
Another suggestion -most of the time a really annoying game mechanic, painful for re-growth a city, especially for AI!:
Peasants conversion to workers priority table is a real pain to the economy. Workers should be spawn-prioritized at best:

Highest spawn priority: Inns, food-productive, breweries workers,
Modest spawn priority: stone quarries & iron mines & ox workers, woodcutters,
Low spawn priority: workshops
Lowest spawn priority: Mills workers and water pots workers(beacuse 3 peasants for these - slows down the recovery the most).

The original spawn priority is always not in place and slows down everything hella much, when it comes to recovery from a peasants lose after an attack, if the economy is large. Especially Inns and breweries, these seem like to come last, which is a real pain, beacuse these are like the most important buildings for recover, they're giving popularity, aswell food, but nope: the AI and your own peasants tend to prioritize workshops, quarries, iron mines... Please change that in the next patch. :P

**Comment by J0J0Jo** [comment]
So if it burns, water pot workers will spawn last?

**Comment by Rakso69** [comment]
@J0J0Jo Good point. My view isn't 100% accurate of course. I hope @Sh0wdown will find the best solution by himself. All we do is just tips. :)

**Comment by Lolasik011** [comment]
Let's hope for @Sh0wdown

**Comment by Sh0wdown** [comment]
Although I would also like this, I think it's too complex for me to implement.
The only thing I could look after is that workers are prioritized before troops, but I don't know if that's a good idea.

**Comment by Heroesflorian** [comment]
The current worker priorities can also be seen (in reverse) when one runs below 50 popularity, as it is reflected in the order in which workers disappear. As such, those priorities are same for AI and human players, so I suppose some code for this must exist already. No idea how difficult it is to find and modify it accordingly, though.

Regarding troops vs workers, this currently seems to come down mostly to timing of recruitment order, as there is a bit of delay between the point where newly spawned peasants can be used for recruitment and the point in time from which they can turn into workers.
